src/legacy/core_plugins/embeddable_api/public/triggers/attach_action.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IRegistry, Trigger } from '../types';
```

Comment:
this method will be consumed from outside (part of our API), it should not require the registry to be passed in but import it internally.

Reply:
I think this all depends on how this ends up fitting into New Platform semantics.  If it's a stateless helper function that can be imported statically, then registries should be passed in.  If it's a function that can only be accessed from the plugin contract, it can use internal registries.  

I agree using internal registries will be a better API.  I just want to kind of do all this in a second pass to clean it up to conform to NP standards.